### ISSUE TITLE
remove automatic complete

### ISSUE THREAD

**Issue by hantek** [hantek/NeuroBricks#16]
the _init_complete() method is causing serious problems, do have it removed.

Consider to use a build() method which is required to be called after the model is defined. The method can recursively call a method in each individual layer class. 

Each method in a class should be inherited independently, i.e., there should be no super() in any method of a subclass which is calling a parent class method with a different name. That makes things cleaner, less likely to hide weird bugs.

**Comment by hantek** [comment]
The build() method should go through all computational nodes in the forward pass, and write intermediate variables in an appropriate place, thus ensures all nodes has a callable name at the script level.

The backward pass is related to how you train the model, so the corresponding training class should take care of the backward pass. 

Also, if there is dropout, batch normalization, etc. added in the network, which changes the structure of the network, it should be written before the build() method. 

(If needed, consider spliting nonlinearities and matrix product in a layer object.)
